### PR TITLE
fix: updated isEmpty util logic

### PR DESCRIPTION
**Pull request by KirankumarAmbati**
Current implementation fails for following conditions:

* Empty string with a space
* Empty object
* Empty array

Screenshot 2022-07-08 at 7 29 09 PM

### CODE REVIEW

**Comment by KirankumarAmbati** [comment]
**CC:** @harish-sethuraman

File: src/utils/index.ts [hunk 1]
```diff
@@ -11,7 +11,11 @@ export const hexToRGBA = (hex: string, alpha: string | number) => {
 };
 
 export const isEmpty = (value: any) =>
-    value === undefined || value === null || value === '' || value === {} || value === [];
+    value === undefined ||
+    value === null ||
+    (typeof value === 'string' && value.trim() === '') ||
+    (Array.isArray(value) && value.length === 0) ||
+    (typeof value === 'object' && Object.keys(value).length === 0);
```
Comment (tripurari001) [inline src/utils/index.ts:18]: Can you handle these edge cases as well ?

Screenshot 2022-07-08 at 8 48 57 PM

**Review by tripurari001 — CHANGES_REQUESTED** [review]

File: src/utils/index.ts (hunk 1 above)
Comment (KirankumarAmbati) [reply src/utils/index.ts]: Added those cases. Just curious: Why aren't we using `lodash.isEmpty()`?

**Review by tripurari001 — APPROVED** [review]
LGTM

File: src/utils/index.ts (hunk 1 above)
Comment (tripurari001) [reply src/utils/index.ts]: > Why aren't we using lodash.isEmpty()

image

**Review by mittalchirag — APPROVED** [review]

**Pull request merged**